### PR TITLE
BUG: ArrowExtensionArray compared to invalid object not raising

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added an entry in the latest `doc/source/whatsnew/v1.5.1.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by jbrockmendel** [comment]
would it make sense to have a code check for returning an exception rather than raising one?  (I think i wrote some methods a while back that intentionally return exceptions to be re-raised by callers which we'd have to change)

**Comment by mroeschke** [comment]
> would it make sense to have a code check for returning an exception rather than raising one?

Did you mean raising exceptions rather than returning them? If so yeah it would help avoid cases like this and https://github.com/pandas-dev/pandas/pull/48834. `try/except` should be more reliable way to re-raise exceptions (specifically) rather than using `return`?

File: pandas/tests/extension/test_arrow.py [hunk 1]
```diff
@@ -1669,6 +1669,12 @@ def test_compare_array(self, data, comparison_op, na_value, request):
                 with pytest.raises(type(exc)):
                     ser.combine(other, comparison_op)
 
+    def test_invalid_other_comp(self, data, comparison_op):
+        with pytest.raises(
```
Comment (phofl) [inline pandas/tests/extension/test_arrow.py:1673]: Can you add gh ref? Otherwise lgtm

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
Thx @mroeschke

**Pull request merged**